contracts/src/v0.8/ccip/test/pools/GHO/GhoTokenPoolEthereum.t.sol

```diff
@@ -0,0 +1,996 @@
+
+    uint256 balancePre = s_token.balanceOf(OWNER);
+    s_token.approve(address(s_ghoTokenPool), amount);
+
+    s_ghoTokenPool.provideLiquidity(amount);
+
+    assertEq(s_token.balanceOf(OWNER), balancePre - amount);
+    assertEq(s_token.balanceOf(address(s_ghoTokenPool)), amount);
+  }
+
+  // Reverts
+
+  function test_UnauthorizedReverts() public {
+    vm.startPrank(STRANGER);
+    vm.expectRevert(abi.encodeWithSelector(Unauthorized.selector, STRANGER));
+
+    s_ghoTokenPool.provideLiquidity(1);
+  }
+
+  function testFuzz_ExceedsAllowance(uint256 amount) public {
+    vm.assume(amount > 0);
+
+    vm.expectRevert(stdError.arithmeticError);
+    s_ghoTokenPool.provideLiquidity(amount);
+  }
+
+  function testLiquidityNotAcceptedReverts() public {
+    s_ghoTokenPool = new UpgradeableLockReleaseTokenPool(address(s_token), 18, address(s_mockRMN), false, false);
+
+    vm.expectRevert(LockReleaseTokenPool.LiquidityNotAccepted.selector);
+    s_ghoTokenPool.provideLiquidity(1);
+  }
+}
+
+contract GhoTokenPoolEthereum_withdrawalLiquidity is GhoTokenPoolEthereumSetup {
+  function testFuzz_WithdrawalLiquiditySuccess(uint256 amount) public {
+    vm.assume(amount < type(uint128).max);
+
+    uint256 balancePre = s_token.balanceOf(OWNER);
+    s_token.approve(address(s_ghoTokenPool), amount);
+    s_ghoTokenPool.provideLiquidity(amount);
+
+    s_ghoTokenPool.withdrawLiquidity(amount);
+
+    assertEq(s_token.balanceOf(OWNER), balancePre);
+  }
+
+  // Reverts
+
+  function test_UnauthorizedReverts() public {
+    vm.startPrank(STRANGER);
+    vm.expectRevert(abi.encodeWithSelector(Unauthorized.selector, STRANGER));
+
+    s_ghoTokenPool.withdrawLiquidity(1);
+  }
+
+  function testInsufficientLiquidityReverts() public {
+    uint256 maxUint128 = 2 ** 128 - 1;
+
+    s_token.approve(address(s_ghoTokenPool), maxUint128);
+    s_ghoTokenPool.provideLiquidity(maxUint128);
+
+    changePrank(address(s_ghoTokenPool));
+    s_token.transfer(OWNER, maxUint128);
+    changePrank(OWNER);
+
+    vm.expectRevert(LockReleaseTokenPool.InsufficientLiquidity.selector);
+    s_ghoTokenPool.withdrawLiquidity(1);
+  }
+}
+
+contract GhoTokenPoolEthereum_transferLiquidity is GhoTokenPoolEthereumSetup {
+  UpgradeableLockReleaseTokenPool internal s_oldLockReleaseTokenPool;
+
+  uint256 internal s_amount = 100_000_000e18;
+
+  error OnlyCallableByOwner();
+  error BridgeLimitExceeded(uint256 limit);
+
+  function setUp() public virtual override {
+    super.setUp();
+
+    s_oldLockReleaseTokenPool = UpgradeableLockReleaseTokenPool(
+      _deployUpgradeableLockReleaseTokenPool(
+        address(s_token),
+        address(s_mockRMN),
+        address(s_sourceRouter),
+        AAVE_DAO,
+        INITIAL_BRIDGE_LIMIT,
+        PROXY_ADMIN
+      )
+    );
+    deal(address(s_token), address(s_oldLockReleaseTokenPool), s_amount);
+    // write to currentBridged
+    vm.store(address(s_oldLockReleaseTokenPool), bytes32(uint256(12)), bytes32(s_amount));
```

Comment:
small nit - probably easier to use the new `currentBridged` setter, because a change in the storage layout would require updating the slot position.

Reply:
[2708501](https://github.com/aave/ccip/pull/20/commits/2708501bf5b6b6ccaa7c9b5c7fe3d2a9cf536aea)